src/app/auth/login/login.component.ts

```diff
@@ -1,57 +1,75 @@
-import {
-  ChangeDetectionStrategy,
-  Component,
-  WritableSignal,
-  inject,
-  signal,
-} from '@angular/core';
-import { FormBuilder, ReactiveFormsModule, Validators } from '@angular/forms';
+import { NgClass } from '@angular/common';
+import { ChangeDetectionStrategy, Component, WritableSignal, inject, signal } from '@angular/core';
+import { FormBuilder, FormGroup, ReactiveFormsModule, Validators } from '@angular/forms';
 import { Router, RouterLink } from '@angular/router';
-import { FormValidationService } from '../form-validation.service';
 import { AuthService } from '../services/auth.service';
 
 @Component({
   selector: 'app-login',
   standalone: true,
-  imports: [ReactiveFormsModule, RouterLink],
+  imports: [ReactiveFormsModule, RouterLink, NgClass],
   templateUrl: './login.component.html',
   styleUrl: './login.component.scss',
-  changeDetection: ChangeDetectionStrategy.OnPush,
+  changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class LoginComponent {
-  private fb = inject(FormBuilder);
-  router = inject(Router);
-  AuthService = inject(AuthService);
-  FormValidationService = inject(FormValidationService);
+  private readonly fb = inject(FormBuilder);
+  private readonly router = inject(Router);
+  private readonly authService = inject(AuthService);
 
-  loginForm = this.fb.nonNullable.group({
+  public loginForm: FormGroup = this.fb.nonNullable.group({
     email: ['', [Validators.required, Validators.email]],
-    password: ['', [Validators.required, Validators.minLength(6)]],
+    password: ['', [Validators.required, Validators.minLength(6)]]
   });
 
   public errorAuthMessage: string = 'You are not registered in the system.';
-  public emptyDataMessage: string =
-    'Check the correctness of the entered data.';
-  public isEmptyData: WritableSignal<Boolean> = signal(false);
+  public isSubmitted: boolean = false;
   public isErrorAuth: WritableSignal<Boolean> = signal<Boolean>(false);
 
-  onSubmit() {
-    const rawForm = this.loginForm.getRawValue();
+  public isEmptyField(field: string): boolean | undefined {
+    if (field) {
+      return this.loginForm.get(field)?.invalid;
+    }
+    return false;
+  }
+
+  public isDirtyField(field: string): boolean | undefined {
+    if (field) {
+      return this.loginForm.get(field)?.dirty;
+    }
+    return false;
+  }
+
+  public isTouchedField(field: string): boolean | undefined {
+    if (field) {
+      return this.loginForm.get(field)?.touched;
+    }
+    return false;
```

Comment:
Функции конечно не очень и думаю, от них можно будет избавиться, но если уж и писать, то можно сделать аккуратнее
```suggestion
    return this.loginForm.get(field)?.touched ?? false;
```